### PR TITLE
Add search plugin

### PR DESCRIPTION
**Pull request by Jacob1Tm**
Very simple, but maybe someone other than me will find this useful

### CODE REVIEW

File: src/plugins/search/index.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            },
+            {
+                type: ApplicationCommandOptionType.STRING,
+                name: "Search query",
+                description: "What do you want to search?",
+                required: true
+            },
+        ],
+        execute(args) {
+            console.log(args);
```
Comment (lewisakura) [inline src/plugins/search/index.ts:75]:
```suggestion
```

File: src/plugins/search/index.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    name: "Search",
+    authors: [Devs.JacobTm],
+    settings,
+    description: "Generates search links for various search engines.",
+    dependencies: ["CommandsAPI"],
+    commands: [{
+        name: "search",
+        description: "Generates search link for selected engine.",
+        options: [
+            {
```
Comment (MrDiamondDog) [inline src/plugins/search/index.ts:60]: I would make this a setting rather than an argument

File: src/plugins/search/index.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                type: ApplicationCommandOptionType.STRING,
+                name: "Search query",
+                description: "What do you want to search?",
+                required: true
+            },
+        ],
+        execute(args) {
+            console.log(args);
+            const query = args[1].value.replace(" ", "+");
```
Comment (lewisakura) [inline src/plugins/search/index.ts:76]: this is a really naive way of doing this and is vulnerable to querystring injection. use encodeURIComponent instead:
```suggestion
            const query = encodeURIComponent(args[1].value);
```

File: src/plugins/search/index.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        name: "search",
+        description: "Generates search link for selected engine.",
+        options: [
+            {
+                type: ApplicationCommandOptionType.STRING,
+                name: "Search engine",
+                description: "Which search engine do you want to use?",
+                required: true,
+                choices: searchEngineChoices,
+            },
```
Comment (lewisakura) [inline src/plugins/search/index.ts:66]: you should set a default or a setting like it seems like you were planning to do

**Review by lewisakura — CHANGES_REQUESTED** [review]
despite that this most likely will not be accepted since the plugin itself is so simple (generating a link to then click, rather than actually giving the search results) i will give a review to assist in future plugin development

**Comment by Jacob1Tm** [comment]
> despite that this will not be accepted since the plugin itself is so simple (generating a link to then click, rather than actually giving the search results) i will give a review to assist in future plugin development

Yea I didn't really expect this to be merged but thanks for review

**Pull request closed without merging**